src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import duke.command.*;
+import duke.command.FindCommand;
+import duke.exception.DukeException;
+import duke.exception.EmptyDateException;
+import duke.exception.EmptyTodoException;
+import duke.exception.UnknownCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * The Parser class helps to make sense of user commands.
+ *
+ * @author Edric Yeo
+ */
+public class Parser {
+
+    /**
+     * Method that takes in a String of user input and returns a specific
+     * type of command to achieve the task specified.
+     *
+     * @param input The line of text inputted by the user.
+     * @return A Command that performs the task specified by the user.
+     */
+    public static Command parse(String input) throws DukeException {
```

Comment:
Could add the @throws in the JavaDoc for this method since it throws an exception